ros2_batch_job/__main__.py

```diff
@@ -295,6 +299,13 @@ def build_and_test(args, job):
         cmd.append('--cmake-args')
         cmd.extend(cmake_args)
 
+    if compile_on_clang:
+        install_clang_cmd = ['sudo apt-get install --no-install-recommends -y clang;']
+        export_clang_variables_cmd = ['export CC=clang CXX=clang++;']
```

Comment:
Instead of exporting this at the top I think it might be cleaner to inject this as a `--cmake-argument` to ament similar to how the gcov flags are inserted below. 

It will also be a lot more visible in the console logs if that argument is explicitly passed to the invocations instead of onetime hidden in the environment at the top.

A few resources; https://stackoverflow.com/a/50494125/7328782 and https://gitlab.kitware.com/cmake/community/wikis/FAQ#how-do-i-use-a-different-compiler

Reply:
Makes sense, I have switched to setting cmake compiler path args instead